### PR TITLE
add form to submit new trips

### PR DESCRIPTION
**Pull request by MiejkeHoltermann**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const trip = Object.fromEntries(formData);
+    const updatedTrips = [
+      { id: uid(), image: "/placeholder.jpg", ...trip },
+      ...initialTrips,
+    ];
+    setTrips(updatedTrips);
+    console.log(trips);
```
Comment (NicoleHoerner) [inline pages/index.js:20]: I think we should not have any console.logs in our final code.

**Review by NicoleHoerner — CHANGES_REQUESTED** [review]
Very nice!😃
Just the console.log. (but I am not that experienced in reviewing code)

File: public/trips.js [hunk 2]
```diff
@@ -1,73 +1,75 @@
...
   {
     location: "Greece",
-    startDate: "12/09/23",
-    endDate: "02/10/23",
+    startDate: "09/12/23",
+    endDate: "10/02/23",
     image: "/greece.jpg",
-    id: "001",
+    id: uid(),
   },
```
Comment (F-Kirchhoff) [inline public/trips.js:10]: please don't put this file into the public folder. This should only contain elements that are shipped directly to the client. You can create a db folder and put this file in there.

File: pages/index.js [hunk 3]
```diff
@@ -1,11 +1,47 @@
...
+    const trip = Object.fromEntries(formData);
+
+    const startYear = startDate.value.substring(2, 4);
+    const startMonth = startDate.value.substring(5, 7);
+    const startDay = startDate.value.substring(8);
+    const formattedStartDate = `${startDay}/${startMonth}/${startYear}`;
+    const endYear = endDate.value.substring(2, 4);
+    const endMonth = endDate.value.substring(5, 7);
+    const endDay = endDate.value.substring(8);
+    const formattedEndDate = `${endDay}/${endMonth}/${endYear}`;
```
Comment (F-Kirchhoff) [inline pages/index.js:23]: you can write a little helper function called 'parseDate' to use it for the start and end date. By this, you don't multiply your code.

File: pages/index.js [hunk 4]
```diff
@@ -1,11 +1,47 @@
...
+    console.log(formattedStartDate, formattedEndDate);
```
Comment (F-Kirchhoff) [inline pages/index.js:24]: please remove all logs before starting the PR 🕵️

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
This looks already great. I added some suggestions how you can structure your code a bit better. Have a look!

File: pages/index.js [hunk 5]
```diff
@@ -34,9 +25,18 @@ export default function HomePage() {
...
     event.target.reset();
   }
 
+  function formatDate(date) {
+    const year = date.value.substring(2, 4);
+    const month = date.value.substring(5, 7);
+    const day = date.value.substring(8);
+    const formattedDate = `${day}/${month}/${year}`;
+    return formattedDate;
+  }
```
Comment (F-Kirchhoff) [inline pages/index.js:38]: please move this function into a folder called `lib/utils.js` and import it here again (only for best practice reasons). You don't need to repost the PR though.

File: pages/index.js [hunk 6]
```diff
@@ -34,9 +25,18 @@ export default function HomePage() {
       ...initialTrips,
     ];
     setTrips(updatedTrips);
+    console.log(updatedTrips);
```
Comment (F-Kirchhoff) [inline pages/index.js:28]: 🕵️

**Review by F-Kirchhoff — APPROVED** [review]
LGTM, please move the helper function formatDate in a separate file as described. but you can go to the QA afterwards

File: components/TripInputForm.js [hunk 7]
```diff
@@ -24,6 +28,7 @@ export default function TripInputForm({ onSubmit }) {
         type="date"
         id="endDate"
         name="endDate"
+        min={start}
```
Comment (F-Kirchhoff) [inline components/TripInputForm.js:31]: sweet!

**Review by F-Kirchhoff — APPROVED** [review]
LGTM

**Pull request merged**